hathor/verification/verification_service.py

```diff
@@ -101,74 +102,59 @@ def verify_basic(self, vertex: BaseTransaction, *, skip_block_weight_verificatio
         Used by `self.validate_basic`. Should not modify the validation state."""
         match vertex.version:
             case TxVersion.REGULAR_BLOCK:
-                assert isinstance(vertex, Block)
+                assert type(vertex) is Block
```

Comment:
Just add a comment saying that we cannot use `isinstance()` here because we're not interest in inheritance.

Reply:
Done in https://github.com/HathorNetwork/hathor-core/pull/830/commits/6d844b94a9889a236b30f525f03754db04a7b087

Reply:
Thanks!